### PR TITLE
Make the plugin not apply to any .txt file [Recommended version]

### PR DESCRIPTION
**Pull request by ZeeQyu**
This is a pull request that will remove the *.txt file name matching and replace it with matching for common names for files that use this syntax (*requirements.txt and *constraints.txt).

I've provided two pull requests, either of which will solve the problem of documentation .txt files getting red error markings and uneven text coloring as a result of requirements language syntax highlighting being applied to it. This is my recommended alternative, but I've provided an alternative in case you don't like the new string matching.

For more details, please see the commit messages.

### CODE REVIEW

**Comment by meanmail** [comment]
There is a problem. For most users, files with requirements have a name other than `requirements.txt`

**Comment by ZeeQyu** [comment]
Of course users can name their requirements files whatever they want, but I don't agree that they're in the majority. From a quick look-through the trending python projects on github, out of the first 10, 5 of them had requirements files at all, and all of them were named exactly requirements.txt. From the python projects I've seen, none of them have another name for their requirements file. Python users are generally quite good at following standards, even loosely defined ones as what a requirements file should be named.

Besides, don't you think most people have some kind of .txt file that isn't a requirements file in their repository? There's a button that appears when opening a requirements file asking the user if they want to install a plugin for this file type. If you press yes, this plugin is installed. This means it's easy to forget you installed this. Then, when you open any text file, this plugin's formatting is applied. The user doesn't know where to start searching. All they know is that a text file has the wrong formatting, not which plugin is causing it.

However, if we assume the users use the name requirements.txt and only match files close to that name (the wildcards I added), a user might open their oddly-named file ("modules-to-install.txt") and realize that it's not being syntax highlighted. Either they don't think about it (no problem) or they know they've installed a requirements plugin and go searching for it in the settings menu. They find the file types menu requirements entry and realize they can add things.

Simply put, it's easier to realize you need to add a file pattern to the plugin than it is to realize the file pattern is being applied to too many files.

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -15,8 +15,8 @@
                   language="requirements.txt"
                   order="first"
                   name="requirements.txt"
-                  fileNames="requirements.txt"
-                  patterns="*.txt"/>
+                  fileNames="requirements.txt;constraints.txt"
+                  patterns="*requirements.txt;*constraints.txt"/>
```
Comment (stdedos) [inline src/main/resources/META-INF/plugin.xml:19]: I'll also leave my glob-suggestions here to die peacefully as well:
```
requirements*.txt
requirements/*.txt
```
